crates/sui-core/src/overload_monitor.rs

```diff
@@ -289,4 +390,264 @@ mod tests {
+    fn start_executor(
+        execution_rate: f64,
+        mut rx: UnboundedReceiver<Instant>,
+        mut stop_rx: oneshot::Receiver<()>,
+        authority: Arc<AuthorityState>,
+    ) -> JoinHandle<()> {
+        tokio::spawn(async move {
+            let mut interval = interval(Duration::from_secs_f64(1.0 / execution_rate));
+            interval.set_missed_tick_behavior(MissedTickBehavior::Skip);
+            loop {
+                tokio::select! {
+                    Some(start_time) = rx.recv() => {
+                        authority.metrics.execution_rate_tracker.lock().record();
+                        authority.metrics.execution_queueing_latency.report(start_time.elapsed());
+                        interval.tick().await;
+                    }
+                    _ = &mut stop_rx => {
+                        info!("Executor stopping");
+                        return;
+                    }
+                }
+            }
+        })
+    }
+
+    // Helper fundtion to periodically print the current overload info.
+    async fn sleep_and_print_stats(state: Arc<AuthorityState>, seconds: u32) {
+        for _ in 0..seconds {
+            info!(
+                "Overload: {:?}. Shedding percentage: {:?}. Queue: {:?}, Ready rate: {:?}. Exec rate: {:?}.",
+                state.overload_info.is_overload.load(Ordering::Relaxed),
+                state
+                    .overload_info
+                    .load_shedding_percentage
+                    .load(Ordering::Relaxed),
+                state.metrics.execution_queueing_latency.latency(),
+                state.metrics.txn_ready_rate_tracker.lock().rate(),
+                state.metrics.execution_rate_tracker.lock().rate(),
+            );
+            sleep(Duration::from_secs(1)).await;
+        }
+    }
+
+    // Running a workload with consistent steady `generator_rate` and `executor_rate`.
+    // It checks that the dropped requests should in between min_dropping_rate and max_dropping_rate.
+    async fn run_consistent_workload_test(
+        generator_rate: f64,
+        executor_rate: f64,
+        min_dropping_rate: f64,
+        max_dropping_rate: f64,
+    ) {
+        let state = TestAuthorityBuilder::new().build().await;
+
+        let (tx, rx) = unbounded_channel();
+        let (_burst_tx, burst_rx) = unbounded_channel();
+        let total_requests = Arc::new(AtomicU32::new(0));
+        let dropped_requests = Arc::new(AtomicU32::new(0));
+        let load_generator = start_load_generator(
+            generator_rate,
+            tx.clone(),
+            burst_rx,
+            state.clone(),
+            true,
+            total_requests.clone(),
+            dropped_requests.clone(),
+        );
+
+        let (stop_tx, stop_rx) = oneshot::channel();
+        let executor = start_executor(executor_rate, rx, stop_rx, state.clone());
+
+        sleep_and_print_stats(state.clone(), 50).await;
```

Comment:
These tests run for about 50 seconds in order to reach to a steady state. Is this ok?

Reply:
try switching the tests to `#[tokio::test(start_paused)]` - this will use a virtual clock so things should very fast. the `multi_thread` flavor for these tests is probably not what you want

Reply:
Amazing! Thanks for the suggestions. Now I'm able to increase this sleep duration with much faster time duration!